File: src/main/java/com/anod/appwatcher/market/AppDetailsFilter.java
package com.anod.appwatcher.market;

import com.anod.appwatcher.utils.CollectionsUtils;
import com.google.android.finsky.api.model.Document;

/**
 * @author algavris
 * @date 03/03/2017.
 */

class AppDetailsFilter {

    static CollectionsUtils.Predicate<Document> createPredicate() {
        return new CollectionsUtils.Predicate<Document>() {
            @Override
            public boolean test(Document document) {
                return document.getAppDetails() == null;
            }
        };
    }
}

File: src/main/java/com/anod/appwatcher/market/AppDetailsFilter.kt
package com.anod.appwatcher.market

import com.google.android.finsky.api.model.FilterPredicate

/**
 * @author algavris
 * *
 * @date 03/03/2017.
 */

internal object AppDetailsFilter {
    val predicate: FilterPredicate = {
        it?.appDetails != null
    }
}
